src/features/treasureIsland/lib/sandPlotMachine.ts

```diff
@@ -0,0 +1,137 @@
+import { InventoryItemName } from "features/game/types/game";
+import { assign, createMachine, Interpreter, State } from "xstate";
+
+export interface SandPlotContext {
+  id: number;
+  discovered?: InventoryItemName | null;
+  dugAt?: number;
+}
+
+export type SandPlotState = {
+  value:
+    | "loading"
+    | "idle"
+    | "digging"
+    | "noShovel"
+    | "treasureFound"
+    | "treasureNotFound"
+    | "opacityTransition"
```

Comment:
Consider naming this something that doesn't leak the implementation details.

`disappearing` or `finishing`